src/Meshes/latlong.jl

```diff
@@ -0,0 +1,34 @@
+using IntervalSets
+
+struct RegularLatLongMesh{S <: Domains.SphereDomain, R <: StepRangeLen} <:
+       AbstractMesh2D
+    domain::S
+    lat::R
+    long::R
+end
+
+function RegularLatLongMesh(
+    domain,
+    nlat::Int,
+    nlong::Int,
+    lat_bounds = (-π / 2, π / 2),
+    long_bounds = (-π, π),
```

Comment:
```suggestion
    lat_bounds = (-π / 2, π / 2),
    long_bounds = (-π, π),
```
we use degrees for bounds everywhere else